src/stores/AppStore/AppStore.ts

```diff
@@ -2108,6 +2111,34 @@ export class AppStore {
         autorun(() => {
             this.activateStatsPanel(this.preferenceStore.statsPanelEnabled);
         });
+
+        // listen devicePixelRatio
+        let remove = null;
+        const updatePixelRatio = () => {
+            if (remove != null) {
+                remove();
+            }
+            const mqString = `(resolution: ${window.devicePixelRatio}dppx)`;
+            const media = matchMedia(mqString);
+            media.addEventListener("change", updatePixelRatio);
+            remove = () => {
+                media.removeEventListener("change", updatePixelRatio);
+            };
+
+            this.pixelRatio.previous = this.pixelRatio.current;
```

Comment:
Do we also consider `imageRatio` here?

Reply:
@loveluthien We also need actions for updating observables.

Reply:
I multiply imageRatio in frameStore to avoid fov problem when export high resolution png